RadixWallet/Features/DerivePublicKeys/DerivePublicKeys.swift

```diff
@@ -204,16 +205,23 @@ extension DerivePublicKeys {
 		loadMnemonicPurpose: SecureStorageClient.LoadMnemonicPurpose,
 		state: State
 	) async throws -> Action {
-		let hdKeys = try await deviceFactorSourceClient.publicKeysFromOnDeviceHD(.init(
-			deviceFactorSource: deviceFactorSource,
-			derivationPaths: derivationPaths,
-			loadMnemonicPurpose: loadMnemonicPurpose
-		))
-		return .delegate(.derivedPublicKeys(
-			hdKeys,
-			factorSourceID: deviceFactorSource.id.embed(),
-			networkID: networkID
-		))
+		do {
+			let hdKeys = try await deviceFactorSourceClient.publicKeysFromOnDeviceHD(.init(
+				deviceFactorSource: deviceFactorSource,
+				derivationPaths: derivationPaths,
+				loadMnemonicPurpose: loadMnemonicPurpose
+			))
+			return .delegate(.derivedPublicKeys(
+				hdKeys,
+				factorSourceID: deviceFactorSource.id.embed(),
+				networkID: networkID
+			))
+		} catch {
+			if error is FailedToFindFactorSource {
+				_ = await overlayWindowClient.scheduleAlert(.missingMnemonicAlert)
+			}
+			throw error
```

Comment:
Should we rethrow even if we show alert? Or only rethrow inside an `else`?

Reply:
Add a `FIXME`, relying on `error is ExpectedErrorType` is not the best code IMO. If we change the error thrown to something else, this breaks, without compiler informing us... It is OK for now, as a quickfix, but I want something more structured in place for the future (soon) which does not break when error type changes.

Reply:
The error still needs to be propagated further so the flow proceeds accordingly, the alert is "just" an intermediary info we show to the user.

As for a more robust solution, I think we talked in one of our standups that we should have a single/reusable slide up that reads from the keychain the mnemonic, that will also by implication show this alert. Currently, as you see, I had to add this check in several places.